src/sentry/static/sentry/app/views/asyncView.jsx

```diff
@@ -1,21 +1,82 @@
 import DocumentTitle from 'react-document-title';
+import idx from 'idx';
+import Modal from 'react-bootstrap/lib/Modal';
 import React from 'react';
 
 import AsyncComponent from '../components/asyncComponent';
+import Button from '../components/buttons/button';
+import ConfigStore from '../stores/configStore';
+import {ApiForm, PasswordField} from '../components/forms';
+import {t} from '../locale';
 
-class AsyncView extends AsyncComponent {
-  constructor(...args) {
-    super(...args);
-  }
-
+export default class AsyncView extends AsyncComponent {
   getTitle() {
     return 'Sentry';
   }
+
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      needsVerification: false,
+    };
+  }
+
+  handleError(error, ...args) {
+    if (idx(error, _ => _.responseJSON.needsSuperuser)) {
+      // we should prompt somehow and maintain loading state
+      return this.setState({
+        needsVerification: true,
+      });
+    } else {
+      return super.handleError(error, ...args);
+    }
+  }
+
+  renderVerifyModal() {
+    let user = ConfigStore.get('user');
+    let location = this.props.location || {pathname: ''};
+    return (
+      <Modal show={this.state.needsVerification} animation={false}>
+        <div className="modal-header">
+          <h4>{t('Confirm Your Identity')}</h4>
+        </div>
+        <div className="modal-body">
+          {user.hasPasswordAuth ? (
```

Comment:
im not sure if we should really be supporting superusers which don't have a password, **except** we intend to remove this requirement by saying "this organization is where authentication for superusers comes from"